spec/support/acceptance/helpers.rb

```diff
@@ -0,0 +1,131 @@
+require 'puppet_litmus'
+# rubocop:disable Style/BracesAroundHashParameters
+
+# The Target class and TargetHelpers module are a useful ways
+# for tests to reuse Litmus' helpers when they want to do stuff
+# on nodes that may not be the current target host (like e.g.
+# the master or the ServiceNow instance).
+#
+# NOTE: This is definitely hacky and should be revisited
+# once puppet-litmus is more polished or we have a better
+# understanding of its design.
+
+class Target
+  include PuppetLitmus
+
+  attr_reader :uri
+
+  def initialize(uri)
+    @uri = uri
+  end
+end
+
+class TargetNotFoundError < StandardError; end
+module TargetHelpers
+  def on_master
+    on_target('master', 'acceptance:provision_vms', 'is_master') do |master|
+      yield master
+    end
+  end
+  module_function :on_master
+
+  def on_servicenow_instance
+    on_target('ServiceNow instance', 'acceptance:setup_servicenow_instance', 'is_servicenow_instance') do |servicenow_instance|
+      yield servicenow_instance
+    end
+  end
+  module_function :on_servicenow_instance
+
+  def on_target(name, setup_task, classifying_var)
```

Comment:
Per the convo with David, this `on_<target>` stuff should probably just be `<target>`, thus avoiding the use of the block.

Reply:
Helper should also take in a `role`, so each target should have a `var['roles']` that we check.